src/main/java/ch/njol/skript/sections/EffSecFireworkLaunch.java

```diff
@@ -0,0 +1,175 @@
+
+		public Firework getFirework() {
+			return firework;
+		}
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+
+	}
+
+	static {
+		Skript.registerSection(EffSecFireworkLaunch.class, "(launch|deploy) [[a] firework [with effect[s]]] %fireworkeffects% at %locations% [([with] (duration|power)|timed) %-number%]");
+		EventValues.registerEventValue(FireworkSectionLaunchEvent.class, Firework.class, new Getter<Firework, FireworkSectionLaunchEvent>() {
+			@Override
+			public Firework get(FireworkSectionLaunchEvent fireworkLaunchEvent) {
+				return fireworkLaunchEvent.getFirework();
+			}
+		}, EventValues.TIME_NOW);
+	}
+
+	private Expression<FireworkEffect> effects;
+	private Expression<Location> locations;
+
+	@Nullable
+	public static Entity lastSpawned;
+
+	@Nullable
+	private Expression<Number> power;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult,
+			@Nullable SectionNode sectionNode, @Nullable List<TriggerItem> triggerItems) {
+
+		effects = (Expression<FireworkEffect>) exprs[0];
+		locations = (Expression<Location>) exprs[1];
+		power = (Expression<Number>) exprs[2];
+
+		if (sectionNode != null)
+			trigger = loadCode(sectionNode, "firework launch", FireworkSectionLaunchEvent.class);
+
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected TriggerItem walk(Event event) {
+		FireworkEffect[] effects = this.effects.getArray(event);
+		// Uses org.bukkit consumer for older versions support. If newer, bukkit maps this class to java utils.
+		Consumer<Firework> consumer = null;
+		if (trigger != null) {
+			consumer = firework -> {
+				FireworkSectionLaunchEvent fireworkLaunchEvent = new FireworkSectionLaunchEvent(firework);
+				Variables.setLocalVariables(fireworkLaunchEvent, Variables.copyLocalVariables(event));
+				TriggerItem.walk(trigger, fireworkLaunchEvent);
+				Variables.setLocalVariables(event, Variables.copyLocalVariables(fireworkLaunchEvent));
+				Variables.removeLocals(fireworkLaunchEvent);
+			};
+		}
+
+		int power = this.power != null ? this.power.getOptionalSingle(event).orElse(1).intValue() : 1;
+		for (Location location : locations.getArray(event)) {
+			World world = location.getWorld();
+			if (world == null)
+				continue;
+			@SuppressWarnings("deprecation")
+			Firework firework = EntityData.spawn(location, Firework.class, consumer);
+			FireworkMeta meta = firework.getFireworkMeta();
+			meta.addEffects(effects);
+			meta.setPower(power);
+			firework.setFireworkMeta(meta);
+			lastSpawned = firework;
+		}
+
+		return super.walk(event, false);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "launch fireworks " + effects.toString(event, debug) +
+				" at " + locations.toString(event, debug) +
+				" timed " + (power != null ? power.toString(event, debug) : "1");
```

Comment:
```suggestion
				" with duration " + (power != null ? power.toString(event, debug) : "1");
```
I think duration is a more familiar term for fireworks